Framework/Algorithms/src/CreateWorkspace.cpp

```diff
@@ -154,39 +172,51 @@ void CreateWorkspace::exec() {
   MatrixWorkspace_const_sptr parentWS = getProperty("ParentWorkspace");
   MatrixWorkspace_sptr outputWS;
   if (parentWS) {
-    // if parent is defined use it to initialise the workspace
-    outputWS =
-        WorkspaceFactory::Instance().create(parentWS, nSpec, xSize, ySize);
+    outputWS = create<HistoWorkspace>(*parentWS, nSpec, histogram);
   } else {
-    // otherwise create a blank workspace
-    outputWS =
-        WorkspaceFactory::Instance().create("Workspace2D", nSpec, xSize, ySize);
+    auto storageMode = Parallel::fromString(getProperty("ParallelStorageMode"));
+    IndexInfo indexInfo(nSpec, storageMode, communicator());
+    outputWS = create<Workspace2D>(indexInfo, histogram);
   }
 
   Progress progress(this, 0.0, 1.0, nSpec);
+  const auto &indexInfo = outputWS->indexInfo();
 
   PARALLEL_FOR_IF(Kernel::threadSafe(*outputWS))
   for (int i = 0; i < nSpec; i++) {
     PARALLEL_START_INTERUPT_REGION
 
+    // In an MPI run the global index i is not necessarily on this rank, i.e.,
+    // there might not be a corrsponding workspace index.
+    const auto localIndices =
+        indexInfo.makeIndexSet({static_cast<GlobalSpectrumIndex>(i)});
+    if (localIndices.empty())
+      continue;
+
     const std::vector<double>::difference_type xStart = i * xSize;
     const std::vector<double>::difference_type xEnd = xStart + xSize;
     const std::vector<double>::difference_type yStart = i * ySize;
     const std::vector<double>::difference_type yEnd = yStart + ySize;
+    auto local_i = localIndices[0];
```

Comment:
No, since there is exactly 1 global index `i`. Locally the result is thus either empty (and we hit `continue` above) or of size 1.